eosio.system/src/delegate_bandwidth.cpp

```diff
@@ -265,7 +266,10 @@ namespace eosiosystem {
          eosio_assert( asset(0) <= tot_itr->net_weight, "insufficient staked total net bandwidth" );
          eosio_assert( asset(0) <= tot_itr->cpu_weight, "insufficient staked total cpu bandwidth" );
 
-         set_resource_limits( receiver, tot_itr->ram_bytes, tot_itr->net_weight.amount, tot_itr->cpu_weight.amount );
+         int64_t ram_bytes, net, cpu;
+         get_resource_limits( receiver, ram_bytes, net, cpu );
+
+         set_resource_limits( receiver, ram_bytes, tot_itr->net_weight.amount, tot_itr->cpu_weight.amount );
```

Comment:
The second argument should instead be `std::max( ram_bytes, tot_itr->ram_bytes + ram_gift_bytes )`.

This allows existing accounts that haven't yet received the RAM gift to trigger it with a simple `delegatebw` or `undelegatebw` action. But it is also resilient to a future system contract that may lower the `ram_gift_bytes` value, which ideally should never happen. In that case, the `std::max` protects against the possibility of insufficient RAM preventing the `delegatebw` and `undelegatebw` actions from completing.